_posts/2019-02-20-snowplow-android-tracker-v1.0.0-released/index.md

```diff
@@ -0,0 +1,134 @@
+3. [Session transitions](#session-transitions)
+4. [Updates](#updates)
+5. [Documentation](#docs)
+6. [Getting help](#help)
+
+<!--more-->
+
+<h2 id="async-support">1. Async support</h2>
+
+There are new methods available for loading the tracker asynchronously (since session information is stored in a file, and must be loaded whenever the tracker starts):
+
+* `waitForSessionFileLoad()`, a blocking call on the tracker to wait for the session file to load.
+
+* `getHasLoadedFromFile()`, a non-blocking call to check whether the tracker has loaded the session file.
+
+* `getLoadFromFileFuture()`, a `Future` that can be used to check the progress.
+
+{% highlight java %}
+// First initialize your tracker
+Tracker.init(new Tracker
+  .TrackerBuilder(e2, "myNamespace", "myAppId", getContext())
+  .base64(false) // Optional - defines if we use base64 encoding
+  .platform(DevicePlatforms.Mobile) // Optional - defines what platform the event will report to be on
+  .subject(new Subject.SubjectBuilder().build()) // Optional - a subject which contains values appended to every event
+  .build()
+);
+
+// Then this method will return a boolean representing if the tracker has loaded
+Tracker.getHasLoadedFromFile()
+{% endhighlight %}
+
+<h2 id="okhttp">2. Share `OkHttpClient` instance</h2>
+
+In this release, emitters can be created with a shared `OkHttpClient` instance.
+
+This practice should save resources, since it will allow sharing the `ConnectionPool` and `ThreadPool` of the client.
+
+Here's an example of the new `Emitter` builder method `client()`:
+{% highlight java %}
+Emitter e2 = new Emitter
+        .EmitterBuilder("com.collector.acme", Context context) // Required
+        .client(sharedOkHttpClient) // Use this for the shared OkHttpClient
+        .build();
+{% endhighlight %}
+
+<h2 id="session-transitions">3. Session transitions</h2>
+
+The tracker now has a builder method `sessionCallbacks()` for supplying session transition `Runnable` callbacks.
+
+These are to be executed whenever there's a transition in the state of the app (from background to foreground or vice versa), as well as whenever there is a session timeout (in the background or foreground).
+
+<h2 id="updates">4. Other changes</h2>
+
+Other updates and fixes include:
+
+* Fix Travis license issue ([#288][288])
+* Update to Android Target API level 28 ([#287][287])
+* Output maximum logging detail in demo app ([#279][279])
+* Perform blocking operations in the background ([#159][159])
+* Enforce strict mode for demo app ([#278][278])
+* Bump versions in build.gradle for demo app ([#284][284])
+* Bump Android Gradle to 3.2.1 ([#280][280])
+* Bump versions of build tools in Travis ([#283][283])
+* Add setter for session context boolean ([#282][282])
+* Update to Android Target API level 26 ([#273][273])
+* Add ability to momentarily suspend session checking ([#271][271])
+
+Many thanks to [Ahmed Khalil][ahmed] for your contributions!
+
+<h2 id="docs">5. Documentation</h2>
```

Comment:
I feel like we should provide a bit more about our motivation here, i.e. helping engineers to get started, making the journey clearer, providing a better path to more in depth documentation.

Reply:
I'm assuming this release includes the documentation overhaul with this comment...